zerver/actions/streams.py

```diff
@@ -93,6 +93,25 @@ def do_deactivate_stream(stream: Stream, *, acting_user: UserProfile | None) ->
 
     ChannelEmailAddress.objects.filter(realm=stream.realm, channel=stream).update(deactivated=True)
 
+    Realm.objects.filter(id=stream.realm_id).update(
+        moderation_request_channel=Case(
+            When(moderation_request_channel=stream, then=Value(None)),
+            default=F("moderation_request_channel"),
+        ),
+        new_stream_announcements_stream=Case(
+            When(new_stream_announcements_stream=stream, then=Value(None)),
+            default=F("new_stream_announcements_stream"),
+        ),
+        signup_announcements_stream=Case(
+            When(signup_announcements_stream=stream, then=Value(None)),
+            default=F("signup_announcements_stream"),
+        ),
+        zulip_update_announcements_stream=Case(
+            When(zulip_update_announcements_stream=stream, then=Value(None)),
+            default=F("zulip_update_announcements_stream"),
+        ),
+    )
```

Comment:
This needs to send an event too, right? I think this needs to call the `do_set_realm_stream` helper instead.

Reply:
Yeah, reworked to send event.

> I think this needs to call the do_set_realm_stream helper instead.

We won't want to perform 4 db operations & send separate events. The updated approach performs 1 update query and sends [`realm op: update_dict` event](https://zulip.com/api/get-events#realm-update_dict).